backend/src/routes/spotify.js

```diff
@@ -0,0 +1,156 @@
+const express = require('express');
+const SpotifyWebApi = require('spotify-web-api-node');
+const crypto = require('crypto');
+const { firebase } = require('../firebase-init.js');
+const { authMiddleware } = require('./auth');
+
+const db = firebase.firestore();
+require('dotenv').config();
+
+const Spotify = new SpotifyWebApi({
+  clientId: process.env.SPOTIFY_CLIENT_ID,
+  clientSecret: process.env.SPOTIFY_CLIENT_SECRET,
+  redirectUri: 'https://hopscotch-backend.herokuapp.com/api/spotify/oauth-callback',
+});
+
+// Scopes to request.
+const SPOTIFY_SCOPES = [
+  'user-read-email',
+  'playlist-modify-public',
+  'playlist-modify-private',
+  'playlist-read-private',
+  'playlist-read-collaborative',
+  'user-read-playback-state',
+  'user-read-currently-playing',
+  'user-library-read',
+  'user-top-read',
+];
+
+/*
+-- Request Params --
+- redirectHost (OPTIONAL) -- If you are attempting to run this locally, you can pass in your local server host (e.g. "http://localhost:5000")
```

Comment:
Should we leave the redirectHost up to the user? Related to the comment I left above

Reply:
redirectHost only works if the redirect url actually matches smth that I've manually registered in the Spotify Developer Console, but yea we can just go w the base url thing